storage/src/cache/worker.rs

```diff
@@ -235,28 +238,38 @@ impl AsyncWorkerMgr {
     }
 
     async fn handle_prefetch_requests(mgr: Arc<AsyncWorkerMgr>, rt: &Runtime) {
+        // Max 2 active requests per thread.
+        mgr.prefetch_sema.add_permits(2);
+
         while let Ok(msg) = mgr.prefetch_channel.recv().await {
             mgr.handle_prefetch_rate_limit(&msg).await;
+            let mgr2 = mgr.clone();
 
             match msg {
                 AsyncPrefetchMessage::BlobPrefetch(state, blob_cache, offset, size) => {
+                    let _ = mgr2.prefetch_sema.acquire().await;
                     if state.load(Ordering::Acquire) > 0 {
-                        let _ = rt.spawn(Self::handle_blob_prefetch_request(
-                            mgr.clone(),
-                            blob_cache,
-                            offset,
-                            size,
-                            state.clone(),
-                        ));
+                        rt.spawn(async move {
```

Comment:
Why delegate the fetching data Future into the same tokio runtime rather than directly run to the end of future `handle_prefetch_requests` . Moreover, it seems the runtime `rt` only has a thread, so re-generate a future is efficent ?

Reply:
The current future acts as a dispatcher, which receives requests from the mpmc queue and dispatch them.
So there may be multiple inflight requests.
Currently two inlfight-requests are allowed.